sycl/doc/design/KernelFusionJIT.md

```diff
@@ -313,12 +313,44 @@ computing the private memory size. As range rounding only applies to basic
 kernels (parametrized by a `sycl::range`), local internalization is not affected
 by the range rounding transformation.
 
+### Reductions
+
+Kernel fusion of reductions is partially supported. In order to preserve the
+legality of the fused kernel, i.e., the fact that fused kernel must perform the
+same work as the graph of kernels to be fused, only the fusion of following
+reduction strategies at the time of writing is supported:
+
+- `group_reduce_and_last_wg_detection`
+- `local_atomic_and_atomic_cross_wg`
+- `range_basic`
+- `group_reduce_and_atomic_cross_wg`
+- `local_mem_tree_and_atomic_cross_wg`
+
+Other strategies require implicit inter-work-group synchronization, not
+supported in kernel fusion.
+
+This way, users should not use `sycl::reduction` directly when performing kernel
```

Comment:
We should not recommend the use of `detail::` as the default. Users should first try the standard interface and only fall back to `detail::` with a fixed, supported strategy in case they encounter issues.